includes/utilities.inc

```diff
@@ -134,13 +166,111 @@ EOQ;
     );
   };
   $issues = array_map($map_results, $results);
-  // Grab the PIDs...
-  $get_pid = function($o) {
-    return $o['pid'];
+  return $issues;
+}
+
+/**
+ * Gets all the issues that the given newspaper owns using Solr.
+ *
+ * @param AbstractObject $object
+ *   An AbstractObject representing a Fedora object.
+ *
+ * @return array
+ *   An array with elements of the form of:
+ *     - pid: The unique persistent identifier for the issue.
+ *     - label: A descriptive label for the issue.
+ *     - sequence: The sequence number of the issue, starts at 1.
+ *     - issued: A DateTime object repersenting the date the issue was released.
+ */
+function islandora_newspaper_get_issues_solr(AbstractObject $object) {
+  $parent_field = variable_get('islandora_newspaper_parent_issue_solr_field', 'RELS_EXT_isMemberOf_uri_ms');
+  $date_field = variable_get('islandora_newspaper_solr_date_field', 'RELS_EXT_dateIssued_literal_ms');
+  $sequence_field = variable_get('islandora_newspaper_solr_sequence_field', 'RELS_EXT_isSequenceNumber_literal_ms');
+  // Counter for looping the start.
+  $start = -1;
+  // Number of records to return at once.
+  $rows = 10000;
+  $results = array();
+
+  $solr_build = new IslandoraSolrQueryProcessor();
+  $solr_query = format_string('!parent_field:("info:fedora/!pid" "!pid") AND'
+    . ' RELS_EXT_hasModel_uri_ms:"info:fedora/islandora:newspaperIssueCModel"',
+    [
+      '!parent_field' => $parent_field,
+      '!pid' => $object->id,
+    ]
+  );
+
+  do {
+    $start += 1;
```

Comment:
Regarding the default dates/sequence, (_the _ms issue I mentioned comments before), could a sort of be added to the Solr query? Not needed?